src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2377,6 +2377,10 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 {
                     Error(diagnostics, ErrorCode.ERR_RefAssignmentMustHaveIdentityConversion, expression.Syntax, targetType);
                 }
+                else
+                {
+                    return expression;
+                }
```

Comment:
This place doesn't feel like the right place for the fix. The expectations are that CreateConversion below should do the right thing since we are passing isCast:false. Bypassing that call is likely to have all sorts of undesired side-effects, like SemanticModel misbehaving or certain diagnostics, especially for tuples, not reported. It looks like a special handling of tuples in CreateConversion for the case of identity conversions is the root cause of the problem. We should consider doing something about that special case, or adjusting lowering behavior to account for that special case instead of returning early here.